oracle_fdw.c

```diff
@@ -2480,41 +2544,17 @@ char
 	initStringInfo(&orderedquery);
 	appendStringInfo(&orderedquery, " ORDER BY");
 
-	/*
-	 * Determine whether we can potentially push query pathkeys to the remote
-	 * side, avoiding a local sort.
-	 */
-	foreach(cell, query_pathkeys)
-	{
+	/* If bestpath->pathkeys found, create sort_clause. */
```

Comment:
This block of code seems partly superfluous (`fdwState->usable_pathkeys` is not really needed here, so why bother with it?), partly it is code repetition (repeated call to `find_em_expr_for_rel` and `deparseExpr`.

I think the clean solution would be to add a field `char *order_clause` to `struct OracleFdwState` and fill that in the code block above (moved to `oracleGetForeignPaths`), where you have to call `deparseExpr` anyway.

That way this whole block could be reduced to

    if (fdwState->order_clause)
        appendStringInfoString(&query, fdwState->order_clause);